### PR TITLE
fix: security, circular reference issues, type checking & errors to logger fixes

### PR DESCRIPTION
**Pull request by bardisg**
## PR Description

Fix issues reported by bugsnag and security vulnerabilities.

- replace vulnerable package component-url with new URL
- add function to filter out circular references when objects are stringified
- attempt to fix the SQLite database connection is busy issue in localstorage-retry
- convert non required error handles into logger.error
- added check for null, is function & type of Event 
- check if plugin name is null or undefined
- fix issues with sanity suite Google Analytics 4 sessionId comparison
- ui & mounting improvements in sanity suite
- add polyfill for Number.isNaN

## Notion ticket

[Ticket link](https://www.notion.so/rudderstacks/Fix-issues-reported-by-bugsnag-bf046f2a2d45463393f295854e4830e7?pvs=4)
[Ticket link](https://www.notion.so/rudderstacks/fix-vulnerabilities-related-to-the-component-url-package-96e3eb9848fb468d9f874f6ce871dc3c?pvs=4)

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/core/analytics.js [hunk 1]
```diff
@@ -187,12 +187,14 @@ class Analytics {
   integrationSDKLoaded(pluginName, modName) {
     try {
       return (
+        pluginName &&
+        modName &&
         window.hasOwnProperty(pluginName) &&
         window[pluginName][modName] &&
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js:193]: I think that was cause we were checking window.hasOwnPropert(undefined) but will also add this in the check

[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js:193]: done

File: src/utils/ObjectUtils.js [hunk 2]
```diff
@@ -17,4 +17,27 @@ const mergeDeepRightObjectArrays = (leftValue, rightValue) => {
 const mergeDeepRight = (leftObject, rightObject) =>
   R.mergeDeepWith(mergeDeepRightObjectArrays, leftObject, rightObject);
 
-export { mergeDeepRightObjectArrays, mergeDeepRight };
+const stringifyWithoutCircular = (obj, excludeNull) => {
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js:23]: added better object type check

**Review by MoumitaM — DISMISSED** [review]

File: src/utils/ObjectUtils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/utils/ObjectUtils.js:23]: @bardisg The problem is not just with arrays but with the circular reference check itself.

```
const d = { "e": "f" };

const a = {"b": d, "c": { g: d } };

stringifyWithoutCircular(a);

>>>
'{"b":{"e":"f"},"c":{"g":"[Circular Reference]"}}'
```

[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js:23]: oh! I see I am not checking for the original object only! Fixing now

[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/utils/ObjectUtils.js:23]: MDN has a neat solution,
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#circular_references

[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js:23]: this does not seem to pass the expected test cases

[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js:23]: checking test cases too

[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/utils/ObjectUtils.js:23]: I haven't run the test script but just manually evaluated the inputs and outputs using this function. It works for me.
However, I changed this function to filter the `null` values.

[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js:23]: got coffee now so it should not take much longer :)

[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/utils/ObjectUtils.js:23]: Haha
image

[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/ObjectUtils.js]: done

**Review by saikumarrs — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**